src/sources/generator.rs

```diff
@@ -0,0 +1,195 @@
+use crate::{
+    event::Event,
+    shutdown::ShutdownSignal,
+    topology::config::{DataType, GlobalOptions, SourceConfig, SourceDescription},
+};
+use futures::{
+    compat::Future01CompatExt,
+    future::{FutureExt, TryFutureExt},
+    stream::StreamExt,
+};
+use futures01::{future::Future, stream::iter_ok, sync::mpsc, Sink};
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use tokio::time::interval;
+
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub(self) struct GeneratorConfig {
+    #[serde(default)]
+    sequence: bool,
+    lines: Vec<String>,
+    #[serde(default)]
+    interval: Option<f64>,
```

Comment:
When I initially saw the name of the `interval` option, I though that it is the interval between individual events instead of batches. Maybe it could make sense to consider naming it `batch_interval` to indicate that it is the interval between batches and not events, or actually make it an interval between individual events instead of batches?

It also might be useful to have an option to generate events with random intervals between them instead of fixed ones. These random intervals between subsequent events can be sampled using [`rand_distr::Exp`](https://docs.rs/rand_distr/0.2.2/rand_distr/struct.Exp.html) function, which would result in the total time it takes to generate a batch of events being distributed according to the [Erlang distribution](https://en.wikipedia.org/wiki/Erlang_distribution).

Reply:
I am open to changing the name. Issue #1317 calls is `repeat`.

Regarding random intervals, it sounds useful. I'd prefer to do that in a separate PR though.